test/jdk/java/lang/reflect/Proxy/ClassRestrictions.java

```diff
@@ -52,129 +58,90 @@ interface Bashful {
         void foo();
     }
 
-    public static final String nonPublicIntrfaceName = "java.util.zip.ZipConstants";
-
-    public static void main(String[] args) {
+    static Stream<Arguments> proxyInterfaces() {
```

Comment:
These are cases where IAE is expected so maybe the method source should be badProxyInterfaces to make it clearer that they are expected to fail?